packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -595,6 +628,20 @@ class Navigator extends StatefulWidget {
   /// A list of observers for this navigator.
   final List<NavigatorObserver> observers;
 
+  /// Called by the navigator to generate the semantic route name for
+  /// accessibility frameworks.
+  ///
+  /// By default, will return the empty string for iOS routes. On Android, the
+  /// initial route is named "home".  Subsequent route names are split on "/"
+  /// and the last piece is returned. For example, "home/login" will produce
+  /// a route name of "login."  Trailing and leading slashes are ignored.
+  ///
+  /// See also:
+  ///
+  ///   * [Semantics.route], the semantics flag which controls screen
+  ///     accessibility.
+  final RouteAccessibilityNameCallback generateAccessibility;
```

Comment:
I don't understand how this API would be used. Why would this not just be part of the route's widget tree?

The place where you create the Navigator isn't where you would configure e.g. the app bar contents, so it shouldn't be where you configure the routes.